style_guide.md

```diff
@@ -0,0 +1,217 @@
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+- Question should be called with question(isProxy=true) or question(isProxy=false):
+```
+{
+  type: 'Question',
+  id: 'confirm_dob',
+  question_variants: [
+    {
+      question: question(isProxy=false)
+    },
+    {
+      question: question(isProxy=true)
+    },
+  ],
+}
+```
+- In the question method, when we need to resolve title, we call a new questionTitle method, passing isProxy:
+```
+local question(isProxy) = {
+  title: questionTitle(isProxy)
+  id: 'confirm-date-of-birth',
+  title: title,
+  type: 'General',
+  answers: [
+  {
+    id: 'confirm-date-of-birth-answer',
+    mandatory: true,
+    options: [
+      {
+        label: 'Yes',
+        value: 'Yes',
+       },
+      {
+        label: 'No',
+        value: 'No',
+      },
+    ],
+    type: 'Radio',
+    },
+  ],
+};
+
+```
+- In the `questionTitle` method we would return either of the two strings, based on the `isProxy` variable:
+```
+local questionTitle(isProxy) = (
+  if isProxy == true then {
+  text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
+  placeholders: [
+    placeholders.personNamePossessive,
+  ]} 
+  else 'What was your age on your last birthday?';
+);
+
+```
+- We would then do a similar thing for guidance with a questionGuidance method:
+```
+local questionGuidance(isProxy) = (
+  if isProxy == true then 'Why we ask this question if they are retired or long-term sick or disabled' else 'Why we ask this question if you are retired or long-term sick or disabled';
+);
+
+```
+    
+
+- Block json can be a function if an argument is being passed:
+
+```
+function(region_code, census_month_year_date) {
+  type: 'Question',
+  id: 'arrive-in-country',
+  question_variants: [
+    {
+      question: question(nonProxyTitle),
+      when: [rules.isNotProxy],
+    },
+    {
+      question: question(proxyTitle),
+      when: [rules.isProxy],
+    },
+  ],
+}
+
+```
+
+#### Methods
+
+- We should define a method each time we’re changing a property.
+- Methods can be invoked at top-level or declared inline, depending on the intended use:
+```
+// TOP-LEVEL
+    
+local guidance(region_code, isProxy) = (
+  if region_code == 'GB-WLS' then
+    if isProxy then walesGuidanceProxy else walesGuidanceNonProxy
+  else if isProxy then englandGuidanceProxy else englandGuidanceNonProxy
+);
+
+// INLINE
+
+local radioOptions = if region_code == 'GB-WLS' then walesOption else englandOption;
+
+  ```
+
+### Variants
+
+- Where variants exist, a question method should be used and passed the variables that control the variant, e.g. `isProxy`.
+```
+{
+  type: 'Question',
+  id: 'date-of-birth',
+  question_variants: [
+    {
+      question: question(isProxy=false)
+    },
+    {
+      question: question(isProxy=true)
+    },
+  ],
+}
+```
```

Comment:
This variants content seems to be a duplicate of earlier content.

Reply:
Removed, thanks!👍